internal/pkg/runtime/engine/singularity/process_linux.go

```diff
@@ -220,17 +219,19 @@ func (e *EngineOperations) StartProcess(masterConn net.Conn) error {
 	// Modify argv argument and program name shown in /proc/self/comm
 	name := "sinit"
 
-	argv0str := (*reflect.StringHeader)(unsafe.Pointer(&os.Args[0]))
-	argv0 := (*[1 << 30]byte)(unsafe.Pointer(argv0str.Data))[:argv0str.Len]
-	progname := make([]byte, argv0str.Len)
+	argv0 := unsafe.Slice(unsafe.StringData(os.Args[0]), len(os.Args[0]))
```

Comment:
For some context around the change here, see: https://github.com/golang/go/issues/53003#issuecomment-1964775055